### ISSUE TITLE
Improve health endpoint with missing required conditions

### ISSUE THREAD

**Issue by AntonioMrtz** [AntonioMrtz/SpotifyElectron#283]
## Description

Improve health endpoint checks with initializations done in `lifespan_handler`

## Context

There are required conditions for the app to work properly that are not checked in health endpoint

## How to do it

Add to health endpoint checks that validate the following initialization in `lifespan_handler` method at `__main__`

```
AuthConfig.init_auth_config(
    access_token_expire_minutes=AppAuthConfig.ACCESS_TOKEN_EXPIRE_MINUTES,
    verification_algorithm=AppAuthConfig.VERTIFICATION_ALGORITHM,
    days_to_expire_cookie=AppAuthConfig.DAYS_TO_EXPIRE_COOKIE,
)

DatabaseConnectionManager.init_database_connection(
    environment=environment, connection_uri=connection_uri
)
SongServiceProvider.init_service()
```

* Validate ping to database
* Validate SongServiceProvider is inited, getting the service and checking it exists
* Validate auth configurations are setted 

## Testing

---

__🖐️ Learn [how to CONTRIBUTE](https://antoniomrtz.github.io/SpotifyElectron/CONTRIBUTING), all help is welcomed :)__

__📚 Check out the [project documentation](https://antoniomrtz.github.io/SpotifyElectron/) for more info__

__⭐ Help this project grow by giving it a star if you like it!__

__😵 Trello links are used for traceability, do not ask to join the board before leaving a comment in the related issue__

**Comment by AntonioMrtz** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Improve health endpoint with missing required conditions](https://trello.com/c/a3sTi8l9/560-improve-health-endpoint-with-missing-required-conditions)

**Comment by RobyBen** [comment]
I'll try to solve this issue

**Comment by AntonioMrtz** [comment]
Hi @RobyBen are you still working on this?

**Comment by AntonioMrtz** [comment]
Hi @berkaykrc would you be interested on tackling this?

**Comment by berkaykrc** [comment]
Yes, I would be interested in any python, backend tagged issue, to be honest on Thursday I checked them all so you read my mind 😁 but the spring semester just started, so I cannot be as focused as my previous submissions. If it is okay with you, I could only look at them on weekends, which will be below my average work time.

**Comment by AntonioMrtz** [comment]
Don't worry, you can tackle it whenever it works for you—no need to rush. In the end, we're contributing voluntarily to this project, so no deadlines are enforced. :)

**Comment by berkaykrc** [comment]
Hey, @AntonioMrtz hope you doing well!
I changed my enviroment and installing requirements with project from stratch but in requirements-global.txt the pre-commit==4.0.2 version but there is no version on their repo? Maybe it is 4.2.0 which is latest version which you mistyped?
It looks like they jumped from 4.0.1 to 4.1.0. Can you look into this?
https://github.com/pre-commit/pre-commit/releases

**Comment by AntonioMrtz** [comment]
Hi @berkaykrc that's true. I will check it ou ! Thanks for informing

**Comment by AntonioMrtz** [comment]
It should be working now @berkaykrc ! :)

**Comment by berkaykrc** [comment]
Thanks, I temporally updated to it 4.2.0 while working just did rebase with your changes.
